contracts/Bridge.sol

```diff
@@ -65,263 +27,117 @@ contract Bridge {
     }
 
     struct DepositProposal {
-        uint                  _destinationChainID;
-        uint                  _depositNonce;
-        bytes32               _dataHash;
-        address[]             _yesVotes;
-        address[]             _noVotes;
-        DepositProposalStatus _status;
+        bytes32                  _dataHash;
+        address[]                _yesVotes;
+        address[]                _noVotes;
+        DepositProposalStatus    _status;
     }
 
     struct RelayerThresholdProposal {
-        uint                           _proposedValue;
+        uint256                        _proposedValue;
         mapping(address => bool)       _hasVoted;
         address[]                      _yesVotes;
         address[]                      _noVotes;
         RelayerThresholdProposalStatus _status;
     }
 
-    // chainID => number of deposits
-    mapping(uint => uint) public _depositCounts;
-    // chainID => depositNonce => GenericDepositRecord
-    mapping(uint => mapping(uint => GenericDepositRecord)) public _genericDepositRecords;
-    // chainID => depositNonce => ERC20DepositRecord
-    mapping(uint => mapping(uint => ERC20DepositRecord)) public _erc20DepositRecords;
-    // chainID => depositNonce => ERC721DepositRecord
-    mapping(uint => mapping(uint => ERC721DepositRecord)) public _erc721DepositRecords;
-    // chainID => depositNonce => CentrifugeAssetDepositRecord
-    mapping(uint => mapping(uint => CentrifugeAssetDepositRecord)) public _centrifugeDepositRecords;
-    // chainID => depositNonce => Proposal
-    mapping(uint => mapping(uint => DepositProposal)) public _depositProposals;
-    // destinationChainID => depositNonce => relayerAddress => bool
-    mapping(uint256 => mapping(uint256 => mapping(address => bool))) public _hasVotedOnDepositProposal;
-
-    event GenericDeposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event ERC20Deposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event ERC721Deposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event CentrifugeAssetDeposited(uint indexed destinationChainID, uint indexed depositNonce);
-    event DepositProposalCreated(uint indexed destinationChainID, uint indexed depositNonce, bytes32 indexed dataHash);
-    event DepositProposalVote(uint indexed destinationChainID, uint indexed depositNonce, Vote indexed vote, DepositProposalStatus status);
-    event DepositProposalFinalized(uint indexed destinationChainID, uint indexed depositNonce);
+    // originChainHandlerAddress => number of deposits
+    mapping(address => uint256) public _depositCounts;
+    // originChainHandlerAddress => depositNonce => bytes
+    mapping(address => mapping(uint256 => bytes)) public _depositRecords;
+    // originChainID => originChainHandlerAddress => depositNonce => depositProposal
+    mapping(uint256 => mapping(address => mapping(uint256 => DepositProposal))) public _depositProposals;
```

Comment:
I'm not usre if `originChainHandlerAddress` is needed in the mapping as a key, `originChainID` & `depositNonce` should be sufficient

Reply:
Because `depositNonce`s are generated based off of:

```solidity
uint256 depositNonce = ++_depositCounts[originChainHandlerAddress];
```

You'll have multiple of the same nonces be generated as different handlers are used

Reply:
why not generate it based on total deposits of teh bridge?

Reply:
So if we did it that way we would avoid the collision issue above, however we loose two features:

1. We wouldn't know how many deposits each handler has handled
2. Because the `depositNonce`s would be non-sequential from the handler's perspective, we wouldn't be able to get all the `depositRecord`s each handler has handled from their mappings

I'm not sure if we'd use either of these features, but I figured having the option might serve us well

Reply:
It was decided that `originChainHandlerAddress` should be removed from the mapping to make sense when interacting with Centrifuge

ISS #8